server/runtime/src/main/java/org/infinispan/server/configuration/resp/RespServerConfigurationParser.java

```diff
@@ -0,0 +1,179 @@
+         ParseUtils.requireNoNamespaceAttribute(reader, i);
+         String value = replaceProperties(reader.getAttributeValue(i));
+         Attribute attribute = Attribute.forName(reader.getAttributeName(i));
+         switch (attribute) {
+            case CACHE: {
+               builder.defaultCacheName(value);
+               break;
+            }
+            case NAME: {
+               builder.name(value);
+               break;
+            }
+            case SOCKET_BINDING: {
+               builder.socketBinding(value);
+               builder.startTransport(true);
+               dedicatedSocketBinding = true;
+               break;
+            }
+            case SECURITY_REALM: {
+               securityRealm = value;
+            }
+            default: {
+               ServerConfigurationParser.parseCommonConnectorAttributes(reader, i, serverBuilder, builder);
+            }
+         }
+      }
+      if (!dedicatedSocketBinding) {
+         builder.socketBinding(endpoint.singlePort().socketBinding()).startTransport(false);
+      }
+      while (reader.inTag()) {
+         Element element = Element.forName(reader.getLocalName());
+         switch (element) {
+            case AUTHENTICATION: {
+               parseAuthentication(reader, serverBuilder, builder.authentication().enable(), securityRealm);
+               break;
+            }
+            case ENCRYPTION: {
+               if (!dedicatedSocketBinding) {
+                  throw Server.log.cannotConfigureProtocolEncryptionUnderSinglePort();
+               }
+               parseEncryption(reader, serverBuilder, builder.encryption(), securityRealm);
+               break;
+            }
+            default: {
+               ServerConfigurationParser.parseCommonConnectorElements(reader, builder);
+            }
+         }
+      }
+   }
+
+   private void parseAuthentication(ConfigurationReader reader, ServerConfigurationBuilder serverBuilder, AuthenticationConfigurationBuilder builder, String securityRealmName) {
+      if (securityRealmName == null) {
+         securityRealmName = serverBuilder.endpoints().current().securityRealm();
+      }
+      for (int i = 0; i < reader.getAttributeCount(); i++) {
+         ParseUtils.requireNoNamespaceAttribute(reader, i);
+         String value = reader.getAttributeValue(i);
+         Attribute attribute = Attribute.forName(reader.getAttributeName(i));
+         switch (attribute) {
+            case SECURITY_REALM: {
+               builder.securityRealm(value);
+               securityRealmName = value;
+               break;
+            }
+            default: {
+               throw ParseUtils.unexpectedAttribute(reader, i);
+            }
+         }
+      }
+
+      ParseUtils.requireNoContent(reader);
+      if (securityRealmName == null) {
+         throw Server.log.authenticationWithoutSecurityRealm();
+      }
+      builder.authenticator(new ElytronRESPAuthenticator(securityRealmName));
```

Comment:
check if the security realm supports passwords?

Reply:
@tristantarrant would have to comment on this.

Reply:
It's handled in `EndpointConfigurationBuilder.enableImplicitAuthentication()`

Reply:
@tristantarrant  is that method invoked when explicit authentication is used?